dev/benchmarks/test_apps/stocks/lib/stocks_app.dart

```diff
@@ -0,0 +1,233 @@
+    debugShowSizes: false,
+    debugShowBaselines: false,
+    debugShowLayers: false,
+    debugShowPointers: false,
+    debugShowRainbow: false,
+    showPerformanceOverlay: false,
+    showSemanticsDebugger: false,
+  );
+
+  @override
+  void initState() {
+    super.initState();
+    _routerDelegate = StockRouterDelegate(_state);
+  }
+
+  void _configurationUpdater(StockConfiguration value) {
+    setState(() {
+      _configuration = value;
+    });
+  }
+
+  ThemeData get theme {
+    switch (_configuration.stockMode) {
+      case StockMode.optimistic:
+        return ThemeData(
+          brightness: Brightness.light,
+          primarySwatch: Colors.purple,
+        );
+      case StockMode.pessimistic:
+        return ThemeData(
+          brightness: Brightness.dark,
+          accentColor: Colors.redAccent,
+        );
+    }
+    assert(_configuration.stockMode != null);
+    return null;
+  }
+
+  @override
+  void dispose() {
+    _routerDelegate.dispose();
+    super.dispose();
+
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    assert(() {
+      debugPaintSizeEnabled = _configuration.debugShowSizes;
+      debugPaintBaselinesEnabled = _configuration.debugShowBaselines;
+      debugPaintLayerBordersEnabled = _configuration.debugShowLayers;
+      debugPaintPointersEnabled = _configuration.debugShowPointers;
+      debugRepaintRainbowEnabled = _configuration.debugShowRainbow;
+      return true;
+    }());
+    return _StockConfigurationScope(
+      configuration: _configuration,
+      state: this,
+      child: MaterialApp.router(
+        title: 'Stocks',
+        theme: theme,
+        localizationsDelegates: StockStrings.localizationsDelegates,
+        supportedLocales: StockStrings.supportedLocales,
+        debugShowMaterialGrid: _configuration.debugShowGrid,
+        showPerformanceOverlay: _configuration.showPerformanceOverlay,
+        showSemanticsDebugger: _configuration.showSemanticsDebugger,
+        routeInformationParser: _routeInformationParser,
+        routerDelegate: _routerDelegate,
+      )
+    );
+  }
+}
+
+class _StockConfigurationScope extends InheritedWidget{
+  const _StockConfigurationScope({
+    Key key,
+    this.state,
+    this.configuration,
+    Widget child,
+  }) : super(key: key, child: child);
+
+  final StockConfiguration configuration;
+  final StocksAppState state;
+
+  @override
+  bool updateShouldNotify(_StockConfigurationScope oldWidget) {
+    return oldWidget.configuration != configuration;
+  }
+}
+
+class StockRouteInformationParser extends RouteInformationParser<StockRoutePath> {
+  @override
+  Future<StockRoutePath> parseRouteInformation(RouteInformation routeInformation) {
+    final List<String> segments = routeInformation.routeName.split('?');
+    final String path = segments[0];
+    final String query = segments.length > 1 ? segments[1] : '';
+    final Uri syntheticUrl = Uri(path: path, query: query);
+    if (syntheticUrl.path == '/settings') {
```

Comment:
maybe define these strings as const on the class to reuse them here and in the method below?